### PR TITLE
#1900 Use RTL in notification-panel tests

### PR DESCRIPTION
**Pull request by srikant-ch5**
Fixes: #1900 
 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

**Comment by nmgokhale** [comment]
@srikant-ch5 Add issue number in PR title -

**Comment by srikant-ch5** [comment]
> @srikant-ch5 Add issue number in PR title - Screenshot 2024-04-23 at 9 46 20 AM

Hi @nmgokhale I have updated PR title with issue number.

File: canvas_modules/common-canvas/__tests__/notification-panel/notification-panel-test.js [hunk 1]
```diff
@@ -128,11 +125,14 @@ describe("notification panel renders correctly", () => {
 			</Provider>
 		);
 
-		const notificationPanel = wrapper.find("NotificationPanel");
-		expect(JSON.stringify(notificationPanel.prop("notificationConfig"))).to.equal(JSON.stringify(notificationConfigDefault));
-		expect(JSON.stringify(notificationPanel.prop("messages"))).to.equal(JSON.stringify(notificationMessages));
-		expect(notificationPanel.prop("subPanelData")).to.equal(subPanelData);
-		expect(notificationPanel.prop("closeSubPanel")).to.equal(closePanelFn);
+		expect(getByText("Notifications Panel")).to.exist;
```
Comment (nmgokhale) [inline canvas_modules/common-canvas/__tests__/notification-panel/notification-panel-test.js:128]: This is not really testing the props to check if the `notificationConfigDefault`, `notificationMessages`, `subPanelData` objects are set.

File: canvas_modules/common-canvas/__tests__/notification-panel/notification-panel-test.js [hunk 2]
```diff
@@ -362,44 +374,166 @@ describe("toolbar notification icon state renders correctly", () => {
...
+			canvasParameters.showBottomPanel,
+			canvasParameters.showRightFlyout,
+			toolbarConfig,
+			notificationConfig,
+			contextMenuConfig,
+			canvasController,
+			rerender
+		);
 		// TODO need to fix
 		// notificationIcon = wrapper.find(".toggleNotificationPanel-action");
```
Comment (nmgokhale) [inline canvas_modules/common-canvas/__tests__/notification-panel/notification-panel-test.js:538]: Can you try fixing this TODO comment?

File: canvas_modules/common-canvas/__tests__/notification-panel/notification-panel-test.js [hunk 3]
```diff
@@ -332,19 +328,35 @@ describe("toolbar notification icon state renders correctly", () => {
...
+		expect(container.querySelectorAll(".notification-panel")).to.have.length(1);
 		expect(canvasController.getNotificationMessages().length).to.equal(0);
-		expect(wrapper.find(".notification-panel-empty-message-container")).to.have.length(1);
+		expect(container.querySelectorAll(".notification-panel-empty-message-container")).to.have.length(1);
 
 		canvasController.setNotificationMessages([notificationMessage0]);
-		wrapper.update();
-		expect(wrapper.find(".toggleNotificationPanel-action")).to.have.length(1);
-		expect(wrapper.find(".notification-panel-empty-message-container")).to.have.length(0);
+		createIntlCommonCanvasRTLRerender(
```
Comment (nmgokhale) [inline canvas_modules/common-canvas/__tests__/notification-panel/notification-panel-test.js:336]: Why is another renderer needed here? Is there any `wrapper.update()` equivalent in RTL?

File: canvas_modules/common-canvas/__tests__/_utils_/common-canvas-utils.js [hunk 4]
```diff
@@ -110,6 +110,88 @@ export function createIntlCommonCanvas(
...
+				showRightFlyout={showRightFlyout}
+				showBottomPanel={showBottomPanel}
+				canvasController={canvasController}
+			/>
+		</IntlProvider>
+	);
+	return wrapper;
+}
+
+export function createIntlCommonCanvasRTLRerender(
```
Comment (nmgokhale) [inline canvas_modules/common-canvas/__tests__/_utils_/common-canvas-utils.js:153]: What is the difference between `createIntlCommonCanvasRTL` and `createIntlCommonCanvasRTLRerender` methods?

Comment (srikant-ch5) [reply canvas_modules/common-canvas/__tests__/_utils_/common-canvas-utils.js:153]: Difference between` createIntlCommonCanvasRTL` and `createIntlCommonCanvasRTLRerender` is that we are passing rerender as argument for `createIntlCommonCanvasRTLRerender`

File: canvas_modules/common-canvas/__tests__/notification-panel/notification-panel-test.js (hunk 1 above)
Comment (nmgokhale) [reply canvas_modules/common-canvas/__tests__/notification-panel/notification-panel-test.js:128]: @srikant-ch5 We are going to remove `enzyme` and `enzyme-adapter-react-17` completely from our codebase. However we have a lot of unit tests to verify props of the component. I did a simple google search and found this article - https://akoskm.com/how-to-test-props-in-react-testing-library

Can you try `jest.mock` and other ways mentioned in the article to test props in RTL?

Comment (srikant-ch5) [reply canvas_modules/common-canvas/__tests__/notification-panel/notification-panel-test.js:128]: Hi @nmgokhale I tried above solution yesterday but once I mocked the component it was affecting other test cases, will try to find a solution for this.

File: canvas_modules/common-canvas/__tests__/notification-panel/notification-panel-test.js (hunk 3 above)
Comment (srikant-ch5) [reply canvas_modules/common-canvas/__tests__/notification-panel/notification-panel-test.js:336]: Sure, I will try to replace it with act().

Comment (nmgokhale) [reply canvas_modules/common-canvas/__tests__/notification-panel/notification-panel-test.js:350]: @srikant-ch5 Couple of issues with this code - 

1. When I try to run just this test by adding `only`, it fails. 
2. `createIntlCommonCanvasRTLRerender()` method is calling `<CommonCanvas / >` again. Earlier test wasn't doing this but it was waiting for `canvasController.setNotificationMessages([notificationMessage0]);` this config to take effect before checking the next code. I think using `async await` here will make the code much cleaner. 
Found this sample code here - https://medium.com/@AbbasPlusPlus/react-testing-library-understanding-act-and-when-to-use-it-301bd06fd1bc
```
import { render } from '@testing-library/react';

it('should fetch and display data', async () => {
  const { findByText } = render(<AsyncComponent />);
  
  const message = await findByText('Hello');
  expect(message).toBeInTheDocument();
});
```

**Review by nmgokhale — APPROVED** [review]
Looks good

**Pull request merged**